### PR TITLE
Create docker image with Mantid static analysis tools

### PR DESCRIPTION
**Pull request by DavidFair**
Creates a docker image which only contains the static analysis tools
required by Mantid.
This decouples our static analysis pipeline from the builders and deps, so we can update each seperately and deploy them to different machines.

### CODE REVIEW

File: static_analysis/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#######
+# cppcheck build
+######
+
+FROM alpine:latest AS cppcheck_build
+
+# Version 1.90
+ARG cppcheck_commit=077e652
+
+RUN \
```
Comment (DanNixon) [inline static_analysis/Dockerfile]: Doesn't this (empty line after `RUN`) generate a warning when building? It definitely used to.

File: static_analysis/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# Download and setup
+WORKDIR /build
+RUN git clone --branch ${LLVM_TAG} --depth 1 https://github.com/llvm/llvm-project.git
+WORKDIR /build/llvm-project
+RUN mv clang llvm/tools
+RUN mv libcxx llvm/projects
+
+# Build
+WORKDIR llvm/build
+RUN cmake -GNinja -DLLVM_BUILD_STATIC=ON -DLLVM_ENABLE_LIBCXX=ON ..
```
Comment (DanNixon) [inline static_analysis/Dockerfile]: Given that `jenkins/jnlp-slave` is Debian based is there a reason we are not just using the pre-built binaries for Kitware?

**Comment by DavidFair** [comment]
Note to self: need to add doxygen

File: static_analysis/Dockerfile (hunk 1 above)
Comment (DavidFair) [reply static_analysis/Dockerfile]: No warning, but it's wrong and a hangover from a restructuring I did mid-way

File: static_analysis/Dockerfile (hunk 2 above)
Comment (DavidFair) [reply static_analysis/Dockerfile]: Debian stretch's version of `clang-format` is too old, I can't seem to find Kitware's PPA
Could you post a link to it as I'd rather drop this if possible

**Comment by DavidFair** [comment]
This should be ready for review again

**Comment by DavidFair** [comment]
Need to add the latest version of CMake yet....

File: static_analysis/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+USER root
+
+RUN apt-get update
+RUN apt-get install -t stretch-backports cmake -y --no-install-recommends
+
+# Ordered in probability of each updating from least->most
+RUN apt-get install clang-format-6.0 -y --no-install-recommends
+RUN apt-get install cmake -y --no-install-recommends
+RUN apt-get install doxygen -y --no-install-recommends
+RUN apt-get install python3 python3-pip python3-setuptools -y --no-install-recommends
```
Comment (DanNixon) [inline static_analysis/Dockerfile]: What is setuptools used for?

Comment (DavidFair) [reply static_analysis/Dockerfile]: Was from me prototyping some other packages and got left behind :+1:

**Comment by DavidFair** [comment]
Sorry about the push spam, I noticed / fixed an issue where cppcheck would fall over since it couldn't find its rules at runtime

**Comment by DavidFair** [comment]
Okay thanks for the reviews so far, I've:
- Added a build / push script
- Added a new deploy script (not tested locally, but it's copy paste of existing with a new image name)
- Tidied the `jenkins-node` subdir a bit

**Comment by martyngigg** [comment]
I'll build the image and get it pushed to DockerHub.

**Pull request merged**